lib/java/test-utils/src/main/java/org/enso/test/utils/ContextUtils.java

```diff
@@ -127,8 +131,10 @@ public void resetOut() {
   @Override
   public void close() {
     if (context != null) {
+      var ref = new WeakReference<>(ensoContext());
       context.close();
       context = null;
+      assertGC("EnsoContext can be GCed when context is closed", true, ref);
```

Comment:
- The ultimate goal is to make sure that once `org.graalvm.polyglot.Context` is closed
- all references to `EnsoContext` are closed 
- that's not really the case right now and fixes may be needed all over the place to make that happen